src/types/redis_json.cc

```diff
@@ -230,4 +230,43 @@ rocksdb::Status Json::ArrLen(const std::string &user_key, const std::string &pat
 
   return rocksdb::Status::OK();
 }
+
+rocksdb::Status Json::StrAppend(const std::string &user_key, const std::string &path, const std::string &value,
+                                std::vector<uint64_t> &append_cnt) {
+  auto ns_key = AppendNamespacePrefix(user_key);
+  JsonMetadata metadata;
+  JsonValue json_val;
+  auto s = read(ns_key, &metadata, &json_val);
+  if (!s.ok()) return s;
+
+  auto append_res = json_val.StrAppend(path, value, append_cnt);
+  if (!append_res) return rocksdb::Status::InvalidArgument(append_res.Msg());
+  // need Reverse order
+  std::reverse(append_cnt.begin(), append_cnt.end());
```

Comment:
I am curious why we need reverse here but not in `StrLen`?

Reply:
`jsoncons::jsonpath::json_replace()` look like is reverse order, like `ArrAppend`: 

> 127.0.0.1:6666> json.set doc $ '{"x":[1],"y":[1, 2], "z": {"y": [1]}}'
OK
127.0.0.1:6666> json.arrappend doc $..y 1
> 1) (integer) 2
> 2) (integer) 3